Enable typeguard in CI (#6458)
numba/numba: docs/source/developer/contributing.rst

    $ python -m numba.runtests --log

> To enable :ref:`runtime type-checking <type_anno_check>`, set environment

Reviewer: ```suggestion
To enable :ref:`runtime type-checking <type_anno_check>`, set the environment
```

> To enable :ref:`runtime type-checking <type_anno_check>`, set environment

Reviewer: Should we also tell people how to get/install typeguard and add it to the list of optional packages in the dependency list (this seems like a similar use case to e.g. `graphviz` for CFG debugging)?

Author: yes, i'll add that

> variable ``NUMBA_USE_TYPEGUARD=1`` and use the `runtests.py` from the source root

Reviewer: ```suggestion
variable ``NUMBA_USE_TYPEGUARD=1`` and use `runtests.py` from the source root
```

exercise all the applicable tests on all the combinations noted above on real
hardware!

.. _type_anno_check:

Type annotation and runtime type checking
'''''''''''''''''''''''''''''''''''''''''

Numba is slowly gaining type annotations. To facilitate the review of pull

> requests that adds typing annotations incrementally, the test suite uses

Reviewer: ```suggestion
requests that are incrementally adding type annotations, the test suite uses
```

> `typeguard`_ to perform runtime type checking, which helps verify the validity

Reviewer: ```suggestion
`typeguard`_ to perform runtime type checking. This helps verify the validity
```

of type annotations.

> To enable runtime type checking in the test suite, users can use the

Reviewer: ```suggestion
To enable runtime type checking in the test suite, users can use
```
